evm_arithmetization/src/cpu/kernel/asm/memory/transient_storage.asm

```diff
@@ -133,26 +143,25 @@ sys_tstore_charge_gas:
     %stack 
         (addr, original_value, slot, value, kexit_info) -> 
         (value, original_value, addr, slot, original_value, kexit_info)
-    EQ %jumpi(sstore_noop)
+    EQ %jumpi(tstore_noop)
 
+add_to_journal:
     // stack: addr, slot, original_value, kexit_info
-global debug_journal:
     %journal_add_transient_storage_change
 
     // stack: kexit_info
     EXIT_KERNEL
 
 new_transient_storage_len:
     // Store the new (unscaled) length.
-    // stack: addr, original_value, slot, value, kexit_info
-    PUSH @SEGMENT_TRANSIENT_STORAGE
-    PUSH 1
-    SUB // 1 - seg
-    ADD // new_len = (addr - seg) + 1
+    // stack: pos, addr, original_value, slot, value, kexit_info
```

Comment:
Just to see if I understood, this part of the code was not correct (because `pos` was not considered part of the stack) but it never got executed because `pos'' > @GLOBAL_METADATA_TRANSIENT_STORAGE_LEN` was always false?

Reply:
No the stack comment was wrong, but we were indeed using `pos` (as we should have) to update the length.
The stack before the conditional jump to this section is `pos'' > len, pos'', addr, original_value, slot, value, kexit_info`, effectively consuming the boolean on top and leaving `pos''` as top of the stack.

Reply:
Yes, I meant that the top of the stack was `pos''` and not `addr`.